src/traces/funnel/hover.js

```diff
@@ -25,16 +25,18 @@ module.exports = function hoverPoints(pointData, xval, yval, hovermode) {
     var di = cd[index];
 
     var sizeLetter = isHorizontal ? 'x' : 'y';
-
     point[sizeLetter + 'LabelVal'] = di.s;
 
-    // display ratio to initial value
-    point.extraText = [
-        formatPercent(di.begR, 1) + ' of initial',
-        formatPercent(di.difR, 1) + ' of previous',
-        formatPercent(di.sumR, 1) + ' of total'
-    ].join('<br>');
-    // TODO: Should we use pieHelpers.formatPieValue instead ?
+    var hoverinfo = trace.hoverinfo;
+    if(hoverinfo !== 'none' && hoverinfo !== 'skip') {
+        var parts = (hoverinfo || '').split('+');
+        var isAll = (hoverinfo === 'all') || (hoverinfo === undefined);
+        var hasFlag = function(flag) { return isAll || parts.indexOf(flag) !== -1; };
+
+        if(hasFlag('percentInitial')) point.percentInitial = formatPercent(di.begR, 1);
+        if(hasFlag('percentPrevious')) point.percentPrevious = formatPercent(di.difR, 1);
+        if(hasFlag('percentTotal')) point.percentTotal = formatPercent(di.sumR, 1);
```

Comment:
Here,

`point.percentIntial = /* should be raw value */`

`point.percentInitialLabel = /* should have the default formatted value */`

`point.extraText = /* should be filled in with percentInitialLabel friends when corresponding hoverinfo flag is present */`

Reply:
I think `sunburst` has got this right:

https://github.com/plotly/plotly.js/blob/3e3b22dc87d9806ec2d1b82e8d29f572263ce155/src/traces/sunburst/plot.js#L556-L584